src/pages/collection/Collection.tsx

```diff
@@ -127,68 +123,67 @@ function Collection() {
   });
+  const [groups, groupsTotal] = useFlattenListResult(groupsQuery.data);
+  const lastPageIds = useMemo(
+    () => groupsQuery.data?.pages.toReversed()[0].List.map(group => group.IDs.ID) ?? [],
+    [groupsQuery],
+  );
 
-  const seriesQuery = useGetFilteredGroupSeriesQuery(
+  const seriesQuery = useFilteredGroupSeries(
+    toNumber(groupId!),
     {
-      groupId: groupId!,
-      randomImages: showRandomPoster,
       filterCriteria: getFilter(debouncedSeriesSearch),
+      randomImages: showRandomPoster,
     },
-    { skip: !isSeries },
+    isSeries,
   );
 
   const isFetching = useMemo(
     () => (isSeries ? seriesQuery.isFetching : groupsQuery.isFetching),
     [isSeries, seriesQuery.isFetching, groupsQuery.isFetching],
   );
-  const [pages, total] = useMemo(
+  const [items, total] = useMemo(
     () => {
-      const data = isSeries ? seriesQuery.currentData : groupsQuery.currentData;
-      return [data?.pages ?? {}, data?.total ?? 0];
+      const data = isSeries ? seriesQuery.data : groups;
+      const tempTotal = isSeries ? seriesQuery.data?.length : groupsTotal;
+      return [data ?? [], tempTotal ?? 0];
     },
-    [isSeries, groupsQuery.currentData, seriesQuery.currentData],
+    [groups, groupsTotal, isSeries, seriesQuery.data],
   );
 
   useEffect(() => {
-    if (!isSeries || debouncedSeriesSearch) return;
-    setTimelineSeries((pages[1] ?? []) as SeriesType[]);
-  }, [debouncedSeriesSearch, isSeries, pages]);
-
-  const [fetchGroupExtras, groupExtrasData] = useLazyGetGroupViewInfiniteQuery();
-  const groupExtras = groupExtrasData.currentData ?? [];
+    if (!isSeries || debouncedSeriesSearch || !seriesQuery.isSuccess) return;
+    setTimelineSeries(seriesQuery.data);
+  }, [debouncedSeriesSearch, isSeries, seriesQuery]);
+
+  // Couldn't find a way to do it in the query itself like we had in RTKQ, so doing it here
+  const [groupExtras, setGroupExtras] = useImmer<WebuiGroupExtra[]>([]);
+  const groupExtrasQuery = useGroupViewQuery({
+    GroupIDs: lastPageIds,
+    TagFilter: 128,
+    TagLimit: 20,
+  });
 
   useEffect(() => {
-    const ids = groupsQuery.data?.pages[currentPage]?.map(group => group.IDs.ID) ?? [];
-    if (!ids.length) return;
-    fetchGroupExtras({
-      GroupIDs: ids,
-      TagFilter: 128,
-      TagLimit: 20,
-    }).then().catch(error => console.error(error));
-  }, [currentPage, fetchGroupExtras, groupsQuery]);
-
-  // This can be inside CollectionView but then defaultPageSize has to be passed to it so same either way
-  // 999 to make it effectively infinite since Group/{id}/Series is not paginated
-  const pageSize = useMemo(() => (groupId ? 999 : defaultPageSize), [groupId]);
+    if (!groupExtrasQuery.isSuccess) return;
+    setGroupExtras(immerState => [...immerState, ...groupExtrasQuery.data]);
+  }, [groupExtrasQuery.data, groupExtrasQuery.isSuccess, setGroupExtras]);
```

Comment:
why is 3rd dependency `setGroupExtras`?

Reply:
Because we are using it on line 177.. Linter wants it

Reply:
That is fine then, I was just thinking maybe I missed something in how those setters work.